pkg/hydrator/nexposeconfig.go

```diff
@@ -0,0 +1,66 @@
+package hydrator
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+
+	httpclient "github.com/asecurityteam/component-httpclient"
+)
+
+// NexposeConfig holds configuration to connect to Nexpose
+// and make a call to the fetch assets API
+type NexposeConfig struct {
+	HTTPClient *httpclient.Config `description:"The HTTP client config from github.com/asecurityteam/component-httpclient."`
+	Host       string             `description:"The scheme and host of a Nexpose instance."`
+	Username   string             `description:"The username used to login to the Nexpose instance at the given host."`
```

Comment:
Suggestion: you leverage the `basicauth` plugin in transportd to remove these values from the code.

Reply:
✔️ I'll create another issue for that.